packages/controls/src/context.ts

```diff
@@ -60,3 +72,71 @@ export type CopyContext = {
 export type MergeContext = {
   mergeElement(a: Element, b: Element): Element
 }
+
+export const ContextResource = {
+  Swatch: 'Swatch',
+  File: 'File',
+  Typography: 'Typography',
+  Table: 'Table',
+  TableColumn: 'TableColumn',
+  Page: 'Page',
+  GlobalElement: 'GlobalElement',
+} as const
+
+export type ContextResource =
+  (typeof ContextResource)[keyof typeof ContextResource]
+
+export function shouldRemoveResource(
+  type: ContextResource,
+  id: string,
+  ctx: CopyContext,
+): boolean {
+  function hasRemoveTag(id: string, map: ResourceMapping): boolean {
+    return removeResourceTagSchema.safeParse(map.get(id)).success
+  }
+
+  switch (type) {
+    case ContextResource.Swatch:
+      return hasRemoveTag(id, ctx.replacementContext.swatchIds)
+    case ContextResource.File:
+      return hasRemoveTag(id, ctx.replacementContext.fileIds)
+    case ContextResource.Typography:
+      return hasRemoveTag(id, ctx.replacementContext.typographyIds)
+    case ContextResource.Table:
+      return hasRemoveTag(id, ctx.replacementContext.tableIds)
+    case ContextResource.TableColumn:
+      return hasRemoveTag(id, ctx.replacementContext.tableColumnIds)
+    case ContextResource.Page:
+      return hasRemoveTag(id, ctx.replacementContext.pageIds)
+    case ContextResource.GlobalElement:
+      return hasRemoveTag(id, ctx.replacementContext.globalElementIds)
```

Comment:
I'd go even further and do this: https://github.com/makeswift/makeswift/pull/1021